proxy/src/proxy.rs

```diff
@@ -391,41 +412,68 @@ async fn connect_to_compute(
             time::Duration::from_secs(10)
         };
 
-        // do this again to ensure we have username?
-        node_info.config.set_startup_params(params);
-
-        match connect_to_compute_once(node_info, timeout).await {
-            Ok(res) => return Ok(res),
-            Err(e) => {
-                error!(error = ?e, "could not connect to compute node");
-                if !can_retry_error(&e, num_retries) {
-                    return Err(e);
+        match state {
+            ConnectionState::Invalid(config, err) => {
+                match try_wake(&config, extra, creds).await {
+                    // we can't wake up the compute node
+                    Ok(None) => return Err(err.into()),
```

Comment:
I'm thinking if this is the kind of error that we want to retry? i.e., if the compute node is in the process of being suspended, probably we can wait a few seconds to retry wakeup here...?

Reply:
> if the compute node is in the process of being suspended

This error emits a BadRequest status, so we do retry that.

I'm not sure if we want to retry on other situations. Perhaps we do, but for now we don't. While refactoring, I try to keep the overall logic the same so I'm hesitant to change it right now